src/language/worker.ts

```diff
@@ -3,21 +3,39 @@ import {
   MongoClientOptions
 } from '@mongosh/service-provider-server';
 import { CompletionItemKind } from 'vscode-languageserver/node';
-import { EJSON } from 'bson';
+import { EJSON, Document } from 'bson';
 import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
 import parseSchema = require('mongodb-schema');
 import { parentPort, workerData } from 'worker_threads';
-
-import type { PlaygroundResult, PlaygroundDebug, ShellExecuteAllResult } from '../types/playgroundType';
+import { PlaygroundResult, PlaygroundDebug, ShellExecuteAllResult } from '../types/playgroundType';
 import { ServerCommands } from './serverCommands';
 
-type EvaluationResult = {
+interface EvaluationResult {
   printable: any;
   type: string | null;
-};
+}
+
 type WorkerResult = ShellExecuteAllResult;
 type WorkerError = any | null;
 
+const content = ({ type, printable }: EvaluationResult) => {
+  if (type === 'Cursor' || type === 'AggregationCursor') {
+    return JSON.parse(EJSON.stringify(printable.documents));
+  }
+
+  return typeof printable === 'string'
+    ? printable
+    : JSON.parse(EJSON.stringify(printable));
```

Comment:
This expression is always the same as `JSON.parse(EJSON.stringify(printable))` – if the concern here is performance, maybe it makes sense to check whether `typeof printable !== 'object' || printable === null` instead of `typeof printable === 'string'`?